ql/src/java/org/apache/hadoop/hive/ql/ddl/table/drop/DropTableAnalyzer.java

```diff
@@ -46,9 +48,13 @@ public DropTableAnalyzer(QueryState queryState) throws SemanticException {
 
   @Override
   public void analyzeInternal(ASTNode root) throws SemanticException {
-    String tableName = getUnescapedName((ASTNode) root.getChild(0));
+    TableName qualTabName = getQualifiedTableName((ASTNode) root.getChild(0));
+    String tableName = qualTabName.getTable();
     boolean ifExists = (root.getFirstChildWithType(HiveParser.TOK_IFEXISTS) != null);
     boolean throwException = !ifExists && !HiveConf.getBoolVar(conf, ConfVars.DROP_IGNORES_NON_EXISTENT);
+    //Authorize database for drop table command
+    Database database  = getDatabase(qualTabName.getDb());
+    outputs.add(new WriteEntity(database, WriteType.DDL_NO_LOCK));
```

Comment:
I find when we create table, the `WriteType` is `DDL_SHARED`, can we change it to that?

Reply:
note, in AbstractDropPartitionAnalyzer we only add outputs for partitions.